frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/state-update/vm-settings-tab-state-update.ts

```diff
@@ -162,6 +168,71 @@ const osUpdater = ({ id, prevState, dispatch, getState }: UpdateOptions) => {
   if (!isWindows && windowsTools) {
     dispatch(vmWizardInternalActions[InternalActionType.RemoveStorage](id, windowsTools.id));
   }
+
+  const iCommonTemplates = iGetLoadedCommonData(state, id, VMWizardProps.commonTemplates);
+  const iTemplate = iCommonTemplates && iGetOSTemplate(iCommonTemplates, os);
+  const pvcName =
+    iTemplate && iGetAnnotations(iTemplate).get(`${TEMPLATE_DATAVOLUME_ANNOTATION}/${os}`);
+
+  const iBaseImages = iGetLoadedCommonData(state, id, VMWizardProps.openshiftCNVBaseImages);
+  const iBaseImage =
+    pvcName &&
+    iBaseImages &&
+    iBaseImages
+      .valueSeq()
+      .filter((iPVC) => iGetName(iPVC) === pvcName)
+      .first();
+
+  dispatch(
+    vmWizardInternalActions[InternalActionType.UpdateVmSettingsField](
+      id,
+      VMSettingsField.CLONE_COMMON_BASE_DISK_IMAGE,
+      {
+        isHidden: asHidden(!iBaseImage, VMSettingsField.OPERATING_SYSTEM),
+        value: !!iBaseImage,
+      },
+    ),
+  );
+};
+
+const cloneCommonBaseDiskImageUpdater = ({ id, prevState, dispatch, getState }: UpdateOptions) => {
+  const state = getState();
+  if (
+    !hasVMSettingsValueChanged(prevState, state, id, VMSettingsField.CLONE_COMMON_BASE_DISK_IMAGE)
+  ) {
```

Comment:
also this thing should not fire when isProviderImport - the changes in source could be dangerous

Reply:
done

Reply:
done